baseplate/secrets/fetcher.py

```diff
@@ -148,23 +208,15 @@ def _make_client(self):
         if nonce:
             login_data["nonce"] = nonce
 
-        logger.debug("Logging into Vault.")
+        logger.debug("Obtaining Vault token via aws_ec2 auth.")
         response = self.session.post(
-            urljoin(self.base_url, "/v1/auth/aws-ec2/login"),
+            urljoin(self.base_url, self.auth_path),
```

Comment:
it feels a bit odd to have the client knowing vault API paths and passing them in directly. is it flexible enough to just parameterize the path based on mount point and let the user specify the mount point but not the whole v1/ ... /login part?

Reply:
sure, we can just require mount_point.